### ISSUE TITLE
Sparse categorical cross entropy shape problem using TensorFlow backend

### ISSUE THREAD

**Issue by coryshain** [keras-team/keras#7666]
The implementation of sparse_categorical_cross_entropy in the TensorFlow backend currently flattens the score tensor and reshapes it if the output is rank 3:

~~~~
    output_shape = output.get_shape()
    targets = cast(flatten(target), 'int64')
    logits = tf.reshape(output, [-1, int(output_shape[-1])])
    res = tf.nn.sparse_softmax_cross_entropy_with_logits(
        labels=targets,
        logits=logits)
    if len(output_shape) == 3:
        # if our output includes timesteps we need to reshape
        return tf.reshape(res, tf.shape(output)[:-1])
    else:
        return res
~~~~

Shouldn't the reshaping occur whenever the output is rank >= 3? I'm building a sequence to sequence model with rank 4 output (nested time series) and sparse_categorical_cross_entropy causes a shape mismatch at runtime because the score tensor it returns is rank 1. This doesn't occur with categorical_cross_entropy, and it appears to be specific to TensorFlow (Theano and CNTK implementations of sparse_categorical_cross_entropy don't check output rank before reshaping). I've currently gotten around this by reshaping the output manually in a custom loss function. But it seems like this could be fixed by simply changing ` if len(output_shape) == 3:` to ` if len(output_shape) >= 3:` in the TensorFlow backend.
